### PR TITLE
Fix update to cocoalumberjack 3 and swift 3

### PR DESCRIPTION
**Pull request by hons82**

### CODE REVIEW

**Comment by swisspol** [comment]
Did you run format source script on this diff?

**Comment by hons82** [comment]
No... why?

**Comment by swisspol** [comment]
Oh sorry, I was thinking of another open source project of mine which has such a script!

File: .swift-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.0
```
Comment (swisspol) [inline .swift-version]: What is this file? This should be in a separate commit anyway.

File: GCDWebServer.podspec [hunk 2]
```diff
@@ -35,7 +35,7 @@ Pod::Spec.new do |s|
   
   s.subspec "CocoaLumberjack" do |cs|
     cs.dependency 'GCDWebServer/Core'
```
Comment (swisspol) [inline GCDWebServer.podspec]: Why the change? Why can't it stay at 2.x? What will break if someone is using 2.x but now has to use 3.x because of this GCDWebServer update.

File: .swift-version (hunk 1 above)
Comment (hons82) [inline .swift-version]: This file is used in the upcoming Cocoapods version 1.1.0 to identify the swift version

File: GCDWebServer.podspec (hunk 2 above)
Comment (hons82) [inline GCDWebServer.podspec]: Cocoalumberjack did the switch to swift 3 with the version 3.0.0. 
Some other projects do keep a separate branch for swift 3, so for a limited time u'll have support for both and insome months they'll probably drop swift 2

**Comment by leoderbois** [comment]
Please accept this MR !

File: GCDWebServer/Core/GCDWebServerPrivate.h [hunk 3]
```diff
@@ -92,6 +92,13 @@
 
 #define __GCDWEBSERVER_LOGGING_FACILITY_COCOALUMBERJACK__
 
+/**
+ * Whether async should be used by log messages, excluding error messages that are always sent sync.
+ **/
+#ifndef LOG_ASYNC_ENABLED
+    #define LOG_ASYNC_ENABLED YES
```
Comment (hons82) [inline GCDWebServer/Core/GCDWebServerPrivate.h]: This is related to CocoaLumberjacks change to swift 3.0
I've got an error without this, so I had to add it here in this branch

Comment (hons82) [inline GCDWebServer/Core/GCDWebServerPrivate.h]: Currently I'm using my fork in my projects because I need swift 3...
meanwhile cocoalumberjack did update, so it might be unnecessary anymore... but I had no time to try

Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerPrivate.h]: I see. This cocoalumberjack change should be in a separate PR. If you take it out from this PR I can merge it. Thanks.

Comment (haifengkao) [inline GCDWebServer/Core/GCDWebServerPrivate.h]: Done. Please merge https://github.com/swisspol/GCDWebServer/pull/275

Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerPrivate.h]: The change for `LOG_ASYNC_ENABLED` is still here?

Comment (haifengkao) [inline GCDWebServer/Core/GCDWebServerPrivate.h]: `LOG_ASYNC_ENABLED` is vital, otherwise this library cannot be built at all.

Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerPrivate.h]: OK, so it seems what we need is 2 commits in one PR or 2 PRs of 1 commit, it doesn't really matter:
- 1 to add the `.swift-version` file which is for CocoaPods
- 1 to bump the version of CocoaLumberJack and add `LOG_ASYNC_ENABLED` to it builds

Does that make sense?

(in my projects I always want to have 1 distinct commit per independent change)

**Comment by EricJones89** [comment]
What is the status on this? We too are having to move to swift 3. Has anyone submitted the updated pull request?

**Comment by hons82** [comment]
@EricJones89 I don't think so. I'm still using my fork in our projects

**Comment by swisspol** [comment]
I'd be happy to land this PR but see my comment above about having this PR made of 2 "clean" commits.

**Comment by kdbertel** [comment]
These have been split out into #275, #290, and #291, assuming I didn't completely screw something up.

**Comment by swisspol** [comment]
This is covered by ac788ca906385dd2718e8071de129860c9de3d88 and #275.
